pkg/scheduler/schedule_one.go

```diff
@@ -309,12 +309,25 @@ func (sched *Scheduler) bindingCycle(
 		return status
 	}
 
-	// Any failures after this point cannot lead to the Pod being considered unschedulable.
+	// Any failures after this point cannot lead to the Pod being considered unschedulable and so we create the PodScheduled condition.
 	// We define the Pod as "unschedulable" only when Pods are rejected at specific extension points, and PreBind is the last one in the scheduling/binding cycle.
 	//
 	// We can call Done() here because
 	// we can free the cluster events stored in the scheduling queue sonner, which is worth for busy clusters memory consumption wise.
 	sched.SchedulingQueue.Done(assumedPod.UID)
+	podScheduled := &v1.PodCondition{
+		Type:   v1.PodScheduled,
+		Status: v1.ConditionTrue,
+	}
+	newStatus := assumedPod.Status.DeepCopy()
+	updated := podutil.UpdatePodCondition(newStatus, podScheduled)
+	if updated {
+		if err := util.PatchPodStatus(ctx, sched.client, assumedPod, newStatus); err != nil {
```

Comment:
If we haven't bind the pod, how can we say: hey, let's update the status to scheduled? I think it doesn't make sense.

And I think the complexity sits at `If we bind successfully, but update the status in failure, who will be responsible for populating the status again? I think for scheduler, if bind successfully, we're end now. 

Maybe delegate the condition update to kubelet is not a bad idea but it should be put in the first place?

Reply:
But have no idea whether this will lead to unexpected behaviors like depending on the podScheduled condition, I think we only give the promise that **Pod is scheduled once spec.nodeName is not empty**.

Reply:
@kerthcet 's concern is valid I think. It's because other than default-scheduler, we can have other players that take over a pod's scheduling lifecycle:

- a pod can be created with a non-nil `.spec.nodeName` - in this case, (default-)scheduler isn't involved at all
- a custom scheduler (yunikorn/volcano/etc.) - they have their own logic from the scratch

Reply:
> This PR fixes a bug where the PodScheduled condition was being created by kube-apiserver.

I don't think it's a _bug_ actually.

> Logically, using the same argument, kube-scheduler should create the PodScheduled condition.

I'd say kube-apiserver, scheduler or scheduler's variant, kubelet can jointly maintain the condition. It's doesn't make to declare kube-scheduler should create (or own) the PodScheduled condition.

> What would you suggest now ? How do I go forward?

I'd say this PR is not on the right direction. We should tackle #127508 with merely fixing the right manager field.

Reply:
> What do you think which component should set it ?

I'm not quite sure if the original issue is general to reproduce, or need some particular sequences of steps. But generally, the manager field should be non-empty and set by the component who updates the PodScheduled condition.

Reply:
If a pod is bound to a node, but the Kubelet is unresponsive, should the condition be set? Semantically it seems like the answer should be yes, which means it needs to be set by the API server.

If the condition is set by the API server, it should also check for `nodeName` being set on pod creation, and add the condition in that case.